cni/deployments/kubernetes/install/test/install_cni.go

```diff
@@ -92,25 +87,42 @@ func ls(dir string, t *testing.T) []string {
 	return fileNames
 }
 
-func cp(src, dest string, t *testing.T) {
+func cp(src, dest string, mode int, t *testing.T) {
 	t.Helper()
 	data, err := ioutil.ReadFile(src)
 	if err != nil {
 		t.Fatalf("Failed to read file %v, err: %v", src, err)
 	}
-	if err = ioutil.WriteFile(dest, data, 0644); err != nil {
+	if err = ioutil.WriteFile(dest, data, os.FileMode(mode)); err != nil {
 		t.Fatalf("Failed to write file %v, err: %v", dest, err)
 	}
 }
 
-func rm(dir string, t *testing.T) {
+func rmDir(dir string, t *testing.T) {
 	t.Helper()
 	err := os.RemoveAll(dir)
 	if err != nil {
 		t.Fatalf("Failed to remove dir %v, err: %v", dir, err)
 	}
 }
 
+// Removes Istio CNI's config from the CNI config file
+// Uses docker to execute the script due to required file permissions
+func rmCNIConfig(testNum int, cniConfigFilename string, containerID string, t *testing.T) {
```

Comment:
It seems we currently don't test the case where `chain=false`. We can always add the test case for it later, but if you feel it will not be much work, it would be nice to have it in this PR.

Reply:
Added the integration tests to this PR